WaterLog_Backend/Procedures.cs

```diff
@@ -201,5 +202,141 @@ private string getSegmentStatus(int segmentId)
             var entry = list.Where(inlist => inlist.SegmentsId == segmentId).Last();
             return entry.EventType;
         }
+
+        //Calculates the data points of the wastage based on period
+        public DataPoints<DateTime,double>[] CalculatePeriodWastage(Period timeframe)
+        {
+            switch (timeframe)
+            {
+                case Period.Daily:
+                    return CalculateDailyWastage(_db.SegmentEvents.Where(a => a.EventType == "leak" && a.TimeStamp.Month == DateTime.Now.Month && a.TimeStamp.Day == DateTime.Now.Day && a.TimeStamp.Year == DateTime.Now.Year).GroupBy(b => b.TimeStamp.Hour).ToList());
```

Comment:
restructure all these long lines in a readable manner

Reply:
it was already done PR 28 from Darren